src/nvim/insexpand.c

```diff
@@ -3291,6 +3299,35 @@ static bool get_next_completion_match(int type, ins_compl_next_state_T *st, pos_
   return found_new_match;
 }
 
+static void get_next_bufname_token(void)
+{
+  int flags = p_ic ? CP_ICASE : 0;
+  char *p;
+
+  FOR_ALL_BUFFERS(b) {
+    if (b->b_p_bl) {
+      char *start = get_past_head(b->b_sfname);
+      char *current = start;
+      p = (char *)path_next_component(start);
+      while (true) {
+        int len = (int)(p - current) - (*p == NUL ? 0 : 1);
+        char *dot = strchr(current, '.');
```

Comment:
A small comment above here like `// treat . as a separator, unless it is the first char in a filename`